File: src/programmers/p1/P1_86491.java
package programmers.p1;


import java.util.Arrays;

//프로그래머스 위클리 챌린지
public class P1_86491 {
    public static void main(String[] args) {
        System.out.println(solution(new int[][]{{60, 50}, {30, 70}, {60, 30}, {80, 40}}));
        System.out.println(solution(new int[][]{{14, 4}, {19, 6}, {6, 16}, {18, 7}, {7, 11}}));
        System.out.println(solution(new int[][]{{1, 2}}));
    }

    public static int solution(int[][] sizes) {
        return Arrays.stream(sizes).reduce((a, b) -> new int[]{
                Math.max(Math.max(a[0], a[1]), Math.max(b[0], b[1])), Math.max(Math.min(a[0], a[1]), Math.min(b[0], b[1]))
        }).map(it -> it[0] * it[1]).get();
    }
}

File: src/programmers/p1/P1_86491.kt
package programmers.p1


fun main() {
println(solution(arrayOf(intArrayOf(1,2))))
}
private fun solution(sizes: Array<IntArray>): Int {
    var max = 0
    var max2 = 0
    sizes.forEach {
        max = it[0].coerceAtLeast(it[1]).coerceAtLeast(max)
        max2 = it[0].coerceAtMost(it[1]).coerceAtLeast(max2)
    }
    return max * max2
}